// Webezy.io Generated proto DO NOT EDIT
syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "OtherPackage.proto";
import "google/protobuf/descriptor.proto";
package webezy.SamplePackage.v1;



// [webezyio] - None
message SampleMessage {
	bool ExtendedField = 1 [
		(webezy.SamplePackage.v1.ExtensionMessage.BoolExtend) = false,
		(webezy.SamplePackage.v1.ExtensionMessage.BoolExtend) = true];
	google.protobuf.Timestamp Timestamp = 2;
	int32 Integer = 3 [(webezy.SamplePackage.v1.ExtensionMessage.TestString) = "Test"];
	webezy.OtherPackage.v1.OtherMessage ChildMessage = 4 [(webezy.SamplePackage.v1.ExtensionMessage.testChild) = {name : "Test",aBool : false}];
	map<string, string> test = 5;
	map<string, webezy.OtherPackage.v1.OtherMessage> testNestedMap = 6 [(webezy.SamplePackage.v1.ExtensionMessage.enum_field) = TEST];
	map<string, webezy.SamplePackage.v1.SampleEnum> testEnumMap = 7 [
		(webezy.SamplePackage.v1.ExtensionMessage.list_enum_field) = TEST,
		(webezy.SamplePackage.v1.ExtensionMessage.list_enum_field) = ANOTHER];
}

// [webezyio] - None
message TestChild {
	string name = 1;
	bool aBool = 2;
}

// [webezyio] - None
message ExtensionMessage {
	extend google.protobuf.FieldOptions {
		repeated bool BoolExtend = 55556;
		string TestString = 55557;
		webezy.SamplePackage.v1.TestChild testChild = 55558;
		webezy.SamplePackage.v1.SampleEnum enum_field = 55559;
		repeated webezy.SamplePackage.v1.SampleEnum list_enum_field = 55560;
	}
}

// [webezyio] - None
enum SampleEnum {
	// [webezyio] - None
	UNKNWON = 0;
	// [webezyio] - None
	TEST = 1;
	// [webezyio] - None
	ANOTHER = 2;
}
